.github/workflows/build.yaml

```diff
@@ -0,0 +1,105 @@
+name: Tag, Build and Push Image
+
+on:
+  push:
+    branches:
+    - develop
+    paths-ignore:
+    - 'README.md'
+    - '.github/**'
+  pull_request:
+    branches:
+    - develop
+    paths-ignore:
+    - 'README.md'
+env:
+  VAULT_ADDR: https://clotho.broadinstitute.org:8200
+  VAULT_PATH_GCR: secret/devops/ci/gcp-sa/broad-dsp-gcr-public/gcr-publish-key
+  REGISTRY_HOST: gcr.io
+  GOOGLE_PROJECT: broad-dsp-gcr-public
+  SERVICE_NAME: ${{ github.event.repository.name }}
+jobs:
+  tag-build-push:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code 
+      uses: actions/checkout@v2
+    - name: Get Short Sha
+      id: short-sha
+      run: echo "::set-output name=sha::$(git rev-parse --short=12 HEAD)"
+    - name: Pull Vault Image
+      run: docker pull vault:1.1.0
+    - name: Get Vault Token
+      id: vault-token-get
+      run: |
+        VAULT_TOKEN=$(docker run --rm --cap-add IPC_LOCK \
+          -e "VAULT_ADDR=${VAULT_ADDR}" \
+          vault:1.1.0 \
+          vault write -field token \
+            auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+            secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+        echo ::set-output name=vault-token::$VAULT_TOKEN
+        echo ::add-mask::$VAULT_TOKEN
+    - name: Get Vault Secrets
+      id: vault-secret-get
+      run: |
+        GCR_KEY=$(docker run --rm --cap-add IPC_LOCK \
```

Comment:
This is definitely not a blocker for merging, but we should be trying to move away from having GitHub actions runners pull secrets from Vault directly. More discussion [here](https://broadinstitute.slack.com/archives/CADM7MZ35/p1615848204267100?thread_ts=1615844943.267000&cid=CADM7MZ35).

Hoping we can move to using Terraform to pull secrets from Vault and [add them to repos](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_secret) that need them instead.

Reply:
Thanks, fixed!